src/main/java/world/bentobox/bentobox/listeners/flags/protection/HurtingListener.java

```diff
@@ -75,9 +75,9 @@ else if (Util.isHostileEntity(e.getEntity()))
     private void respond(EntityDamageByEntityEvent e, Entity damager, Flag flag) {
         // Get the attacker
         if (damager instanceof Player player) {
-            checkIsland(e, player, player.getLocation(), flag);
+            checkIsland(e, player, e.getEntity().getLocation(), flag);
```

Comment:
Fixes the exploit where players could attack from an adjacent island where these things were allowed.